src/OMSimulatorLib/ComponentFMUCS.cpp

```diff
@@ -752,7 +775,11 @@ oms_status_enu_t oms::ComponentFMUCS::setBoolean(const ComRef& cref, bool value)
     return logError_UnknownSignal(getFullCref() + cref);
 
   if (oms_modelState_virgin == getModel()->getModelState())
-    startValues.setBoolean(allVariables[j].getCref(), value);
+    if (Flags::ExportParametersInline())
+      startValues.setBoolean(allVariables[j].getCref(), value);
+    else
+      // append startValues with prefix (e.g) addP.K1
+      startValues.setBoolean(getCref()+allVariables[j].getCref(), value);
```

Comment:
I think we should put curly brace around this to make it easier to read:
```C
if (oms_modelState_virgin == getModel()->getModelState())
{
    if (Flags::ExportParametersInline())
      startValues.setBoolean(allVariables[j].getCref(), value);
    else
      startValues.setBoolean(getCref() + allVariables[j].getCref(), value);
}

Reply:
added the curly brackets to improve readability of conditions statements see https://github.com/OpenModelica/OMSimulator/pull/755/commits/c55abe718394f2adc80c6c6ec71331f4b8629098